### PR TITLE
Chapter 5 self test

### PR DESCRIPTION
**Pull request by codavex**

### CODE REVIEW

File: src/ch05/Encode.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		String msg = "This is a test";
+		String encmsg = "";
+		String decmsg = "";
+		String key = "datacash";
+
+		System.out.print( "Original message: " );
+		System.out.println( msg );
+
+		// encode the message 
+		for ( int i = 0 ; i < msg.length() ; i++ )
```
Comment (alexdeas) [inline src/ch05/Encode.java:16]: Using the one loop counter to track positions in both strings was a neat idea. I ended up using a separate counter as I really wanted to change this into an enhanced for.

File: src/ch05/MinMax2.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// Use array initializers.
+class MinMax2 {
+
+	public static void main( String args[] ) {
+		int nums[] = { 99, -10, 100123, 18, -978,
+		    5623, 463, -9, 287, 49 };
+		int min, max;
+
+		min = max = nums[0];
+		// this will now iterate over nums[0] which is unneccesary
```
Comment (alexdeas) [inline src/ch05/MinMax2.java:12]: Unnecessary, but enhanced!

File: src/ch05/self-test.txt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+11. Yes, overruns are errors. Yes, negative values are out of range. 
+
+12. >>>
+
+13. see MinMax2.java
+
+14. No, because you need the positions as well as the values to swap.
+
+15. Yes, since JDK 7.
```
Comment (alexdeas) [inline src/ch05/self-test.txt:31]: Right answers!

File: src/ch05/BubbleString.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		// display original array  
+		System.out.print( "Original array is:" );
+		for ( int i = 0 ; i < size ; i++ )
+			System.out.print( " " + strs[i] );
+		System.out.println();
+
+		// This is the bubble sort.  
+		for ( a = 1 ; a < size ; a++ )
+			for ( b = size - 1 ; b >= a ; b-- ) {
+				if ( strs[b - 1].compareTo( strs[b] ) > 0 ) { // if out of order
```
Comment (SandyThomson) [inline src/ch05/BubbleString.java:27]: This looks a bit odd if you start to have uppercase characters (FWICR). I mixed the use of `compareTo()` and `compareToIgnoreCase()`

File: src/ch05/Encode.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+			encmsg = encmsg + (char) (msg.charAt( i ) ^ key.charAt( i % key.length() ));
```
Comment (SandyThomson) [inline src/ch05/Encode.java:17]: Good stuff, you used modulus to deal with the scenario where your string is longer than your key! :smiley:

File: src/ch05/self-test.txt (hunk 3 above)
Comment (SandyThomson) [inline src/ch05/self-test.txt:31]: I concur!

**Pull request closed without merging**